### PR TITLE
feat(schema): Add structs doc

### PR DESCRIPTION
**Pull request by zmstone**
Support document generation for struct (type)

### CODE REVIEW

**Review by k32 — APPROVED** [review]

File: src/hocon_schema.erl [hunk 1]
```diff
@@ -84,13 +87,17 @@
          , desc => iodata()
          }.
 
+-type struct_meta() :: #{desc => iodata()}.
 -type field() :: {name(), typefunc() | field_schema()}.
+-type fields() :: [field()] | #{fields := [field()],
```
Comment (zhongwencool) [inline src/hocon_schema.erl:92]: There are more meta's to be added than just `desc`, such as: 
I need to set this meta to mark this configuration as hot updatable, generating the HTTP API put method.

File: src/hocon_schema.erl [hunk 2]
```diff
@@ -194,13 +201,23 @@ find_structs(Schema) ->
...
 
+%% @doc Get fields of the struct for the given struct name.
 -spec fields(schema(), name()) -> [field()].
-fields(Mod, Name) when is_atom(Mod) ->
-    Mod:fields(Name);
-fields(#{fields := Fields}, Name) when is_function(Fields) ->
-    Fields(Name);
-fields(#{fields := Fields}, Name) when is_map(Fields) ->
-    maps:get(Name, Fields).
+fields(Sc, Name) ->
```
Comment (zhongwencool) [inline src/hocon_schema.erl:206]: if we only return fields meta by `fields/1`, then the `find_struct/1` can't be private. 
I need the get struct from emqx_dashboard apps to generate API.

so `find_struct/1` should be public or make fields/1 return all meta(bad idea...)

Comment (zmstone) [reply src/hocon_schema.erl:206]: There is fields_and_meta/1

Comment (zmstone) [reply src/hocon_schema.erl:206]: Exported

**Review by zhongwencool — DISMISSED** [review]

**Comment by zmstone** [comment]
rebased

**Pull request merged**